contracts/L2/token/LivepeerToken.sol

```diff
@@ -3,42 +3,54 @@ pragma solidity ^0.8.0;
 
 import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
 import {ERC20, ERC20Permit} from "@openzeppelin/contracts/token/ERC20/extensions/draft-ERC20Permit.sol";
-import {ILivepeerToken} from "./ILivepeerToken.sol";
+import {ERC20Burnable} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
 
-contract LivepeerToken is ILivepeerToken, AccessControl, ERC20Permit {
+contract LivepeerToken is AccessControl, ERC20Burnable, ERC20Permit {
     bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
     bytes32 public constant BURNER_ROLE = keccak256("BURNER_ROLE");
 
+    event Mint(address indexed to, uint256 amount);
+    event Burn(address indexed burner, uint256 amount);
+
     constructor() ERC20("Livepeer Token", "LPT") ERC20Permit("Livepeer Token") {
         _setupRole(DEFAULT_ADMIN_ROLE, _msgSender());
         _setRoleAdmin(MINTER_ROLE, DEFAULT_ADMIN_ROLE);
         _setRoleAdmin(BURNER_ROLE, DEFAULT_ADMIN_ROLE);
     }
 
     /**
-     * @dev Function to mint tokens
+     * @notice Function to mint tokens
+     * @dev Only callable by addreses with MINTER_ROLE
      * @param _to The address that will receive the minted tokens.
      * @param _amount The amount of tokens to mint.
      */
-    function mint(address _to, uint256 _amount)
-        external
-        override
-        onlyRole(MINTER_ROLE)
-    {
+    function mint(address _to, uint256 _amount) external onlyRole(MINTER_ROLE) {
         _mint(_to, _amount);
         emit Mint(_to, _amount);
     }
 
     /**
-     * @dev Burns a specific amount of the sender's tokens
+     * @notice Burns a specific amount of msg.sender's tokens
+     * @dev Only callable by addresses with BURNER_ROLE
+     * @param _amount The amount of tokens to be burned
+     */
+    function burn(uint256 _amount) public override onlyRole(BURNER_ROLE) {
+        super.burn(_amount);
+        emit Burn(msg.sender, _amount);
+    }
+
+    /**
+     * @notice Burns a specific amount of an address' tokens
+     * @dev Only callable by addresses with BURNER_ROLE. Requires the address to approve the caller to burn the amount
+     * @param _from Address to burn tokens for
      * @param _amount The amount of tokens to be burned
      */
-    function burn(address _from, uint256 _amount)
-        external
+    function burnFrom(address _from, uint256 _amount)
+        public
```

Comment:
can make this function external too

Reply:
The contract currently inherits from [ERC20Burnable](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/extensions/ERC20Burnable.sol#L14) because I didn't want to re-implement the `burnFrom()` logic there and that abstract contract sets the visibility of `burnFrom()` to public forcing this declaration to also be declared as public (cannot change visibility when overriding virtual functions).

The alternative here would be to re-implement the `burnFrom()` logic. Doing this just to declare the function as external doesn't seem worth it for me at the moment so I elected to just keep the function as public.

Reply:
ah right